lardataobj/MCBase/MCMiniPart.h

```diff
@@ -0,0 +1,153 @@
+#include "nusimdata/SimulationBase/MCParticle.h"
+#include "nusimdata/SimulationBase/MCTruth.h" // simb::Origin_t
+
+// ROOT libraries
+#include <TVector3.h>
+
+// STL
+//#include <set>
+#include <utility> // std::pair<>
+#include <vector>
+
+#include "TLorentzVector.h"
+
+namespace sim
+{
+
+  class MCMiniPart {
+
+  public:
+
+    /// Default constructor
+    MCMiniPart() {Reset();}
+
+    /// Default destructor
+    //virtual ~MCMiniPart() = default;
+
+    void Reset() { ResetData(); }
+
+    MCMiniPart(MCMiniPart const &) = default; // copy constructor
+    MCMiniPart& operator=( const MCMiniPart &) = default;
+    MCMiniPart(MCMiniPart&&) = default;
+    MCMiniPart& operator= (MCMiniPart&&) = default;
+
+    void ScaleStartMom(double factor) { _start_mom *= factor; }
+    void ScaleEndMom(double factor)  { _end_mom *= factor; }
+    void AddDaughter(unsigned int d) {
+      if (!this->HasDaughter(d))
+        _daughters.push_back(d);
+    }
+    bool HasDaughter(unsigned int d) const {
+      return std::find(_daughters.begin(), _daughters.end(), d) != _daughters.end();
+    }
+
+    // Getters
+    const simb::Origin_t& Origin () const { return _origin; }
+    int            PdgCode() const { return _pdgcode; }
+    unsigned int   TrackID() const { return _track_id; }
+    const std::string&    Process() const { return _process; }
+    unsigned int   Mother()  const { return _mother; }
+    unsigned int   Ancestor() const { return _ancestor; }
+    const TLorentzVector& StartVtx() const { return _start_vtx; }
+    const TLorentzVector& StartMom() const { return _start_mom; }
+    const TLorentzVector& EndVtx()   const { return _end_vtx; }
+    const TLorentzVector& EndMom()   const { return _end_mom; }
+    const std::vector<std::pair<TLorentzVector,TLorentzVector> >& DetPath() const { return _det_path; }
+    const std::vector<unsigned int>& Daughters() const { return _daughters; }
+
+    // Setters
+    void Origin (simb::Origin_t o) { _origin = o; }
+    void PdgCode(int id)           { _pdgcode = id; }
+    void TrackID(unsigned int id)  { _track_id = id; }
+    void Process(const std::string &name) { _process = name; }
+    void Mother(unsigned int id)   { _mother = id; }
+    void Ancestor(unsigned int id) { _ancestor = id; }
+    void StartVtx(const TLorentzVector& vtx) { _start_vtx = vtx; }
+    void StartMom(const TLorentzVector& mom) { _start_mom = mom; }
+    void EndVtx(const TLorentzVector& vtx)   { _end_vtx = vtx; }
+    void EndMom(const TLorentzVector& mom)   { _end_mom = mom; }
+    void DetPath(const std::vector<std::pair<TLorentzVector,TLorentzVector> >& p) { _det_path = p; }
+    void DetPath(std::vector<std::pair<TLorentzVector,TLorentzVector> >&& p) { _det_path = std::move(p); }
+    void Daughters(const std::vector<unsigned int>& d) { _daughters = d; }
+    void Daughters(std::vector<unsigned int>&& d) { _daughters = std::move(d); }
+
+    /// Explicit conversion operator to simb::MCParticle
```

Comment:
Same question as for the `SimEnergyDepositLite` class.  Do you intend this to be an `explicit` conversion operator--i.e. using the `explicit` keyword in front of the `operator` keyword?  Or are you just saying that you're being explicit by specifying a conversion operator in the class?  If the latter, I suggest removing the comment on line 93 as it's clear this is a conversion operator and may cause confusion because there's no `explicit` specified in the conversion-operator function signature.

Reply:
It was the latter, I removed the comment.